pypesto/optimize/optimizer.py

```diff
@@ -282,23 +303,8 @@ def minimize(
                 options=self.options,
             )
 
-        # some fields are not filled by all optimizers, then fill in None
-        grad = getattr(res, 'grad', None) if self.is_least_squares() \
-            else getattr(res, 'jac', None)
-        fval = res_to_chi2(res.fun) if self.is_least_squares() \
-            else res.fun
-
         # fill in everything known, although some parts will be overwritten
         optimizer_result = OptimizerResult(
-            x=res.x,
-            fval=fval,
-            grad=grad,
-            hess=getattr(res, 'hess', None),
```

Comment:
I think this should be kept as `res` may store hessian approximations via BFGS etc that are not available from history.

Reply:
this concerns the general question whether we fill results from history alone, or from the used optimizer. advantage from history: we know what we compute, and format fits. advantage from optimizer: it is (hopefully) ensured that the point is valid, and it can also fill in e.g. bfgs approximations. in the latter case, one could fill from history only if None is provided so far.

Taking your point into account, maybe filling from optimizer is better.

Reply:
Yeah, I would prefer values from history over optimizer results. Extracting when not available from history is probably the best way to go.

Reply:
ah right. also an option. so we only override from history, if a value is provided in the history for that point.

Reply:
No, I would do it the other way around, as you proposed. Only override from optimizer if no value is provided in history.